src/charm.py

```diff
@@ -82,7 +84,12 @@ def private_ip(self) -> str:
     def _on_install(self, event: InstallEvent):
         if not _is_pro_attached():
             self.unit.status = BlockedStatus("Waiting for Ubuntu Pro attachment")
-        self._snap.install()
+            return
+        try:
+            self._snap.install()
+        except Exception:
+            event.defer()
```

Comment:
Will this trigger the _on_install function to be invoked later on?

Reply:
yes this will retrigger the event before any next event occurs.

Reply:
If the same exception keeps being thrown, the above logic would result in the self.snap.install() function being called infinitely?

Reply:
yes until the snap does not get installed since that is a critical path for the charm anyway. An `errored` charm also does the same thing when juju retries the charm through the `update-status` hook. We can also rely on juju doing that.

Reply:
Okay, I was wondering how often this will be triggered. If it works as usual, being triggered every 5 minutes, that would be fine.